spec/03-plugins/37-opentelemetry/04-exporter_spec.lua

```diff
@@ -808,7 +808,7 @@ for _, strategy in helpers.each_strategy() do
         }, { "opentelemetry" }))
 
         setup_instrumentations("all", {
-          endpoint = "{vault://env/test_otel_endpoint}",
+          traces_endpoint = "{vault://env/test_otel_endpoint}",
```

Comment:
does this highlight a breaking change? I tried to run the test with `endpoint` and it seems it now fails indeed with: `schema violation (endpoint: both deprecated and new field are used but their values mismatch: endpoint = {vault://env/test_otel_endpoint} vs traces_endpoint = http://127.0.0.1:21306)`.

This appears to be because the dereferencing is only done when `traces_endpoint` is set, but the compared value (the vault reference) is detected as a difference. This is an issue we already had, but before this change, it was only a problem when the two fields (deprecated and new) were both set, and a vault reference was used as a value.

Now, with this change, the `endpoint` field is dereferenced and copied into the new `traces_endpoint` field, thus resulting in a breaking change for whoever is using a vault reference for the `endpoint` field and upgrades to this version.

cc @nowNick wdyt?

Reply:
> Now, with this change, the endpoint field is dereferenced and copied into the new traces_endpoint field,
Where does this dereferencing happen? Could you please help me by pointing to the code maybe, that does the dereferencing part? Since, I just added a new field in the deprecation block, which already existed, I didn't understand how the issue was caused.

Reply:
That's a good point about references @samugi ! I think adding `replaced_with` to the field itself might not be _"bad"_ - it's supposed to be there so that the endpoint returns responses in backwards compatible manner. However I don't think this approach is going to fix the underlying issue that's been raised in https://github.com/Kong/deck/issues/1404 The problem is that not all deprecations are simple renaming. An example here is `cluster_nodes` in [RateLimitingAdvanced](https://docs.konghq.com/hub/kong-inc/rate-limiting-advanced/configuration/#config-redis-cluster_nodes). It is defined as a list of records: 
`[ {ip1, port2}, {ip2, port2}, ... ]` 
but it used to be an list of strings: 
`[ 'ip1:port1', 'ip2:port2', ... ]`. 

So to translate between those two values (new and old) is not as simple as rewriting this data back. I think we need a different approach when it comes to new & deprecated fields on decK diff.

Reply:
to clarify, @Prashansa-K , I don't think **your change** in particular is problematic, it just highlighted a problem (during schema validation?) that already existed, i.e. if a field is referenceable, adding a `replaced_with` to the corresponding shorthand can create breaking change scenarios for users who had the field configured with a vault reference as a value. In which case, apparently, the plugin's configuration will start throwing schema validation failures.

@nowNick about:
> I think adding replaced_with to the field itself might not be "bad"

I think we should avoid adding `replaced_with` to referenceable fields until the issue above is resolved, wdyt?

Reply:
Yeah. You're probably right @samugi . Could you provide a small decK file that highlights this issue? (probably the one you used for testing?)

Reply:
I didn't actually test this on decK, the test that reproduces this is actually the one that this thread is referencing.